instrumentation/log4j/log4j-appender-2.17/library/src/main/java/io/opentelemetry/instrumentation/log4j/appender/v2_17/internal/LogEventMapper.java

```diff
@@ -169,51 +185,200 @@ void captureMessage(LogRecordBuilder builder, AttributesBuilder attributes, Mess
+      } else if (first instanceof Long) {
+        attributes.put(
+            (ExtendedAttributeKey<List<Long>>)
+                keyProvider.apply(key, ExtendedAttributeType.LONG_ARRAY),
+            (List<Long>) value);
+      } else if (first instanceof Float) {
+        attributes.put(
+            (ExtendedAttributeKey<List<Double>>)
+                keyProvider.apply(key, ExtendedAttributeType.DOUBLE_ARRAY),
+            ((List<Float>) value).stream().map(Float::doubleValue).collect(Collectors.toList()));
+      } else if (first instanceof Double) {
+        attributes.put(
+            (ExtendedAttributeKey<List<Double>>)
+                keyProvider.apply(key, ExtendedAttributeType.DOUBLE_ARRAY),
+            (List<Double>) value);
+      }
+    } else if (value instanceof String[]) {
+      attributes.put(
+          (ExtendedAttributeKey<List<String>>)
+              keyProvider.apply(key, ExtendedAttributeType.STRING_ARRAY),
+          Arrays.asList((String[]) value));
+    } else if (value instanceof boolean[]) {
+      boolean[] arr = (boolean[]) value;
+      List<Boolean> list = new ArrayList<>(arr.length);
+      for (boolean f : arr) {
+        list.add(f);
+      }
+      attributes.put(
+          (ExtendedAttributeKey<List<Boolean>>)
+              keyProvider.apply(key, ExtendedAttributeType.BOOLEAN_ARRAY),
+          list);
+    } else if (value instanceof long[]) {
+      List<Long> list = Arrays.stream((long[]) value).boxed().collect(Collectors.toList());
+      attributes.put(
+          (ExtendedAttributeKey<List<Long>>)
+              keyProvider.apply(key, ExtendedAttributeType.LONG_ARRAY),
+          list);
+    } else if (value instanceof int[]) {
+      List<Long> list =
+          Arrays.stream((int[]) value).mapToLong(i -> i).boxed().collect(Collectors.toList());
+      attributes.put(
+          (ExtendedAttributeKey<List<Long>>)
+              keyProvider.apply(key, ExtendedAttributeType.LONG_ARRAY),
+          list);
+    } else if (value instanceof double[]) {
+      List<Double> list = Arrays.stream((double[]) value).boxed().collect(Collectors.toList());
+      attributes.put(
+          (ExtendedAttributeKey<List<Double>>)
+              keyProvider.apply(key, ExtendedAttributeType.DOUBLE_ARRAY),
+          list);
+    } else if (value instanceof float[]) {
+      float[] arr = (float[]) value;
+      List<Double> list = new ArrayList<>(arr.length);
+      for (float f : arr) {
+        list.add((double) f);
+      }
+      attributes.put(
+          (ExtendedAttributeKey<List<Double>>)
+              keyProvider.apply(key, ExtendedAttributeType.DOUBLE_ARRAY),
+          list);
+    } else if ((value instanceof Map)) {
+      ExtendedAttributesBuilder nestedAttribute = ExtendedAttributes.builder();
+      Map<?, ?> nestedMap = (Map<?, ?>) value;
+      consumeAttributes(
+          consumer -> nestedMap.forEach((k, v) -> consumer.accept(k.toString(), v)),
```

Comment:
Here I convert keys of nested maps to string via `toString`. What should we do if the key is not a string? Is this fine, or should we throw an exception?